engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2272,7 +2329,7 @@ public void deploy(Set<PTContainer> releaseContainers, Collection<PTOperator> un
         for (PTOperator oper : e.getValue()) {
           // operator will be deployed after it has been undeployed, if still referenced by the container
           if (oper.getState() != PTOperator.State.PENDING_UNDEPLOY) {
-            oper.setState(PTOperator.State.PENDING_DEPLOY);
+            oper.setState(PTOperator.State.PENDING_UNDEPLOY);
```

Comment:
Can you please check what happens in the initial for loop that sets all operators to UNDEPLOY? Why isn't the operator marked there?